Modules/itertoolsmodule.c

```diff
@@ -846,7 +787,8 @@ tee_copy(teeobject *to, PyObject *Py_UNUSED(ignored))
 {
     teeobject *newto;
 
-    newto = PyObject_GC_New(teeobject, &tee_type);
+    itertoolsmodule_state *state = itertoolsmodule_find_state_by_type(Py_TYPE(to));
+    newto = PyObject_GC_New(teeobject, state->tee_type);
```

Comment:
You can use `PyObject_GC_New(teeobject, Py_TYPE(to));` directly in here to avoid performance issue.

Reply:
> You can use `PyObject_GC_New(teeobject, Py_TYPE(to));` directly in here to avoid performance issue.

Ah, of course. Thank you!